ci/benchmark_parser.py

```diff
@@ -115,6 +126,18 @@ def parse_key_sizes(result_file):
     return result_values
 
 
+def compute_ops_per_dollar(data_point, product_hourly_cost):
+    """
+    Compute numbers of operations per dollar for a given ``data_point``.
+
+    :param data_point: timing value measured during benchmark
+    :param product_hourly_cost: cost in dollar per hour of hardware used
+
+    :return: number of operations per dollar
+    """
+    return ONE_HOUR_IN_NANOSECONDS / (product_hourly_cost * data_point)
```

Comment:
this formula looks wrong

Let's say I have a cost = 10 $/hour

let's say data_point is the mean time to run a PBS in nanoseconds so let's say we run 1 PBS per nanosecond we have data_point = 1 PBS/ns

cost_per_nanosecond = cost / ONE_HOUR_IN_NANOSECONDS it's expressed in $/ns

and now we can compute our PBS per dollar :

data_point / cost_per_nanosecond = (PBS / ns) / ($ / ns) = (PBS * ns) / ($ * ns) = PBS/$ 

so I think your full formula is currently computing the cost of a PBS instead of the number of PBS per dollar

Reply:
I think I got it right, here's the rationale:

- criterion measure as time per benched function (thus `nanoseconds/PBS`)
- I have hourly cost for AWS (thus  `$/hour`)

Since criterion results are in nanoseconds and we want `PBS/$`, I do:

`one_hour_in_ns (ns/h) / hourly_cost ($/h) = duration_per_dollar (ns/$)`
then
`duration_per_dollar (ns/$) / data_point (ns/PBS) = pbs_per_dollar (PBS/$)`

It seems that my dimensional analysis is right. At least it yields coherent values during parsing.
From what I see, your first hypothesis is wrong a data_point is not PBS per nanosecond but rather nanoseconds per PBS.

Reply:
you are right 😵 I completely mixed the meaning of data point